<?xml version='1.0' encoding='iso-8859-1' standalone='no'?>
<!DOCTYPE issue SYSTEM "lwg-issue.dtd" [
  <!ENTITY nbsp "&#160;">
] >

<issue num="1416" status="C++11">
<title>[FCD] <tt>forward_list::erase_after</tt> should not be allowed to throw</title>
<section><sref ref="[container.requirements]"/></section>
<submitter>DIN</submitter>
<date>25 Aug 2010</date>

<discussion>
<p><b>Addresses DE-21</b></p>

<p>
23.2.1&#47;11 provides a general no-throw guarantee for
erase() container functions, exceptions from this are
explicitly mentioned for individual containers. Because of
its different name, forward_list's erase_after() function is
not ruled by this but should so.
</p>

<note>
Resolved in Rapperswil by a motion to directly apply the words from the ballot comment in N3102.
</note>

</discussion>

<resolution>
<p>
Add a "<i>Throws</i>: Nothing" clause to both
<tt>erase_after</tt> overloads in 23.3.3.4, [forwardlist.modifiers].
</p>
</resolution>

</issue>
